router/dataplane.go

```diff
@@ -2901,15 +2900,26 @@ func (p *scionPacketProcessor) verifyCurrentMAC( /*@ ghost dp io.DataPlaneSpec,
 		// @ }
 		return tmpRes, tmpErr
 	}
+
 	// Add the full MAC to the SCION packet processor,
 	// such that EPIC does not need to recalculate it.
 	p.cachedMac = fullMac
 	// @ reveal p.EqAbsInfoField(oldPkt)
 	// @ reveal p.EqAbsHopField(oldPkt)
-	// (VerifiedSCION) Assumptions for Cryptography:
-	// @ absInf := p.infoField.ToAbsInfoField()
+
 	// @ absHF := p.hopField.ToIO_HF()
-	// @ AssumeForIO(dp.hf_valid(absInf.ConsDir, absInf.AInfo.V, absInf.UInfo, absHF))
+	// @ absInf := p.infoField.ToAbsInfoField()
+
+	// @ unfold acc(sl.Bytes(p.hopField.Mac[:path.MacLen], 0, path.MacLen), R21)
+	// @ unfold acc(sl.Bytes(fullMac[:path.MacLen], 0, path.MacLen), R21)
+	// @ path.EqualBytesImplyEqualMac(fullMac, p.hopField.Mac)
+	// ==> absHF.HVF := path.AbsMac(p.hopField.Mac)
+	//  == path.AbsMac(FromSliceToMacArray(fullMac))
+
+	// Needed to "access" postcondition of `FullMAC`:
+	// @ unfold acc(sl.Bytes(fullMac, 0, len(fullMac)), R21)
+	// ==> absHF.HVF == io.nextMsgtermSpec(...)
```

Comment:
please drop comments like this once you are done with the changes, as they make the code harder to read